datafusion/functions/src/datetime/to_date.rs

```diff
@@ -26,9 +26,42 @@ use datafusion_expr::scalar_doc_sections::DOC_SECTION_DATETIME;
 use datafusion_expr::{
     ColumnarValue, Documentation, ScalarUDFImpl, Signature, Volatility,
 };
+use datafusion_macros::udf_doc;
 use std::any::Any;
 use std::sync::OnceLock;
 
+#[udf_doc(
```

Comment:
this is very cool

@Omega359 what do you think?

Reply:
I dumped the example #[udf_doc ...] into my editor to do a direct comparison to the existing solution and to be honest it definitely is nicer to visually parse. 

There are situations currently where this approach will not work, at least as it currently seems to be implemented. For example:  [bit_and_or_xor](https://github.com/apache/datafusion/blob/main/datafusion/functions-aggregate/src/bit_and_or_xor.rs), some of the regex udfs, etc.